File: 052_Quiz - inheritance/src/hr/unipu/java/EMultipleChoiceQuestion.java
package hr.unipu.java;

// Nasljeđivanje – eliminacija ponavljanja
public class EMultipleChoiceQuestion extends EShortAnswerQuestion  {
    private String correctAnswer2;


    public EMultipleChoiceQuestion(String text, String correctAnswer1, String correctAnswer2) {
        super(text, correctAnswer1);
        //this.correctAnswer1 = correctAnswer1;
        this.correctAnswer2 = correctAnswer2;
    }

//    public String getText() {
//        return text;
//    }

    @Override
    public boolean isCorrectAnswer(String answer) {
        return correctAnswer.equalsIgnoreCase(answer) || correctAnswer2.equalsIgnoreCase(answer);
    }

}

File: 052_Quiz - inheritance/src/hr/unipu/kotlin/EMultipleChoiceQuestion.kt
package hr.unipu.kotlin

// Nasljeđivanje – eliminacija ponavljanja
// -  //this.correctAnswer1 = correctAnswer1;
class EMultipleChoiceQuestion(text: String,
                              correctAnswer1: String,
                              private val correctAnswer2: String)
    : EShortAnswerQuestion(text, correctAnswer1) {

    //    public String getText() {
    //        return text;
    //    }

    override fun isCorrectAnswer(answer: String): Boolean {
        return correctAnswer.equals(answer, ignoreCase = true) || correctAnswer2.equals(answer, ignoreCase = true)
    }

}
